viam-cartographer/src/slam_service/slam_service.cc

```diff
@@ -647,10 +647,31 @@ void SLAMServiceImpl::SaveMapWithTimestamp() {
     }
 }
 
+void SLAMServiceImpl::AddSensorReading(
+    cartographer::sensor::TimedPointCloudData measurement) {
+    cartographer::transform::Rigid3d tmp_global_pose;
+    bool update_latest_global_pose = false;
+
+    {
+        std::lock_guard<std::mutex> lk(map_builder_mutex);
+        if (measurement.ranges.size() > 0) {
+            trajectory_builder->AddSensorData(kRangeSensorId.id, measurement);
+            auto local_poses = map_builder.GetLocalSlamResultPoses();
+            if (local_poses.size() > 0) {
+                update_latest_global_pose = true;
+                tmp_global_pose = map_builder.GetGlobalPose(trajectory_id,
+                                                            local_poses.back());
```

Comment:
What does `local_poses.back()` return?

Reply:
the last element of `local_poses`, its a [method on std::vector](https://cplusplus.com/reference/vector/vector/back/)

Reply:
@kkufieta 
1. are we guaranteed that the most recent pose is always the last element of `local_poses`? 
2. Does this mean that `local_poses` grows unbounded over the lifetime of a given carto_grpc_server process?

Reply:
From what I can tell, the most recent pose is added to the back of `local_pose` and nothing is ever removed from it so yes and yes, but please correct me if I am wrong!

Reply:
1. Yes, I do also believe that the most recent pose is always the last element of `local_poses`.
2. I am not sure, but my guess is "yes" - in the same manner as the overall map (pbstream) is growing unbounded